instrumentation/okhttp/okhttp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/okhttp/v3_0/OkHttp3InstrumentationModule.java

```diff
@@ -37,20 +37,32 @@ public OkHttp3InstrumentationModule() {
   public static class OkHttpClientInstrumentation implements TypeInstrumentation {
     @Override
     public ElementMatcher<TypeDescription> typeMatcher() {
-      return named("okhttp3.OkHttpClient");
+      return named("okhttp3.OkHttpClient$Builder");
     }
 
     @Override
     public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
       return singletonMap(
-          isConstructor().and(takesArgument(0, named("okhttp3.OkHttpClient$Builder"))),
-          OkHttp3InstrumentationModule.class.getName() + "$OkHttp3Advice");
+          isConstructor(), OkHttp3InstrumentationModule.class.getName() + "$OkHttp3Advice");
     }
   }
 
   public static class OkHttp3Advice {
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void addTracingInterceptor(@Advice.Argument(0) OkHttpClient.Builder builder) {
+    public static void trackCallDepth(@Advice.Local("callDepth") int callDepth) {
+      callDepth = CallDepthThreadLocalMap.incrementCallDepth(OkHttpClient.Builder.class);
+    }
+
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void addTracingInterceptor(
+        @Advice.This OkHttpClient.Builder builder, @Advice.Local("callDepth") int callDepth) {
+      if (callDepth > 0) {
+        return;
+      }
```

Comment:
it's not clear why we need the call depth check (since already checking below if interceptor is present before adding), can you add comment explaining?

Reply:
Yeah added